modules/library/opengis/src/main/java/org/opengis/filter/ComplexPropertyAccessor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.opengis.filter;
+
+import org.opengis.feature.Property;
+
+/**
+ * Used for filters to get object properties based on an xpath expression.
+ * 
+ * @author Alvaro Huarte
+ */
+public interface ComplexPropertyAccessor {
```

Comment:
Gak - a new public API that breaks, since only features are made of up Property objects why is this needed?

Could just check if an object is an instanceof Feature?